src/components/widgets-settings/attach-kpi/attach-kpi.component.coffee

```diff
@@ -0,0 +1,95 @@
+module = angular.module('impac.components.widgets-settings.attach-kpis', [])
+module.component('settingAttachKpi', {
+  bindings: {
+    widget: '<'
+    chartPromise: '<?'
+    options: '<?'
+    onInit: '&?'
+    onComplete: '&?'
+  }
+  templateUrl: 'widgets-settings/attach-kpi.tmpl.html'
+  controller: ($timeout, ImpacKpisSvc, ImpacDashboardsSvc)->
+    ctrl = this
+
+    ctrl.$onInit = ->
+      # Define properties
+      # -------
+      ctrl.kpi = {}
+      ctrl.showPanel = false
+      ctrl.draftTarget = ''
+      ctrl.options ||= {
+        chartShrinkSize: 30
+      }
+      ctrl.dhbCurrency = ImpacDashboardsSvc.getCurrentDashboard().currency
+      # Emit API to parent
+      ctrl.onInit($event: { api: { createKpi: ctrl.createKpi } })
+      # Get attachable kpi templates
+      ImpacKpisSvc.getAttachableKpis(ctrl.widget.endpoint).then((kpiTemplates)->
+        # TODO: support for multiple kpi's.
+        angular.extend(ctrl.kpi, angular.copy(kpiTemplates[0]))
+      )
+      # Register to dashboard change for currency update
+      ImpacDashboardsSvc.dashboardUpdated().then(null, null, (dhb)->
+        ctrl.dhbCurrency = dhb.currency
+      )
+      # Register to chart changes
+      ctrl.chartPromise.then(null, null, (chart)->
+        ctrl.chart = chart
```

Comment:
just to make sure I get it right: this is going to be called when the chart is set in the widget?